lib/kafo/param.rb

```diff
@@ -120,12 +120,8 @@ def interpret_validation_args(args)
       args.map do |arg|
         if arg.to_s == "$#{self.name}"
           self.value
-        elsif arg.class.name == 'Puppet::Parser::AST::ASTArray'
-          interpret_validation_args(arg.to_a)
-        elsif arg.class.name == 'Puppet::Parser::AST::Concat'
-          interpret_validation_args(arg.value).join
-        elsif arg.respond_to? :value
-          arg.value
+        elsif arg.is_a? String
+          arg.gsub("$#{self.name}", self.value.to_s)
```

Comment:
I'm struggling to understand why it's not caught by the first if condition, what does the string contains as extra characters? and wouldn't be `sub` instead of `gsub` enough?